### PR TITLE
Copter: enable CONDITION_YAW turn rate in auto

### PR DESCRIPTION
**Pull request by tajisoft**
I want the angular speed of MAV_CMD_CONDITION_YAW(115) to be reflected in auto. I've tested it with SITL, but I still want some advice to make it better.

### CODE REVIEW

**Comment by muramura** [comment]
@tajisoft san. This method is also accessed by MAV_CMD_DO_MOUNT_CONTROL.
Does it take into account the rotation speed of the gimbal in the rotation speed of the fuselage?

File: ArduCopter/autoyaw.cpp [hunk 1]
```diff
@@ -123,12 +123,10 @@ void Mode::AutoYaw::set_fixed_yaw(float angle_deg, float turn_rate_dps, int8_t d
         // default to regular auto slew rate
         _fixed_yaw_slewrate = AUTO_YAW_SLEW_RATE;
     } else {
-        const int32_t turn_rate = (wrap_180_cd(_fixed_yaw - curr_yaw_target) / 100) / turn_rate_dps;
-        _fixed_yaw_slewrate = constrain_int32(turn_rate, 1, 360);    // deg / sec
+        _fixed_yaw_slewrate = wrap_360_cd(turn_rate_dps) * 100;
```
Comment (muramura) [inline ArduCopter/autoyaw.cpp:126]: @tajisoft san.  Yaw rotation can be specified as relative or absolute angle.
In the case of a relative angle, the parameter's angular velocity seems to be sufficient.
However, for absolute angles, the specified angular rate will not be used.
For example, if the angle is currently 90 degrees and the absolute angle is 180 degrees, the movement angle will be 90 degrees.
The planner does not need to know the current angle to make the angle you specify.
I think this PR requires the planner to know the current angle.

File: ArduCopter/autoyaw.cpp [hunk 2]
```diff
@@ -123,12 +123,10 @@ void Mode::AutoYaw::set_fixed_yaw(float angle_deg, float turn_rate_dps, int8_t d
...
     }
 
-    // set yaw mode
-    set_mode(AUTO_YAW_FIXED);
+    set_rate(_fixed_yaw_slewrate);
```
Comment (muramura) [inline ArduCopter/autoyaw.cpp:129]: The original will hold the head in place at the final angle.
I don't think this change fixes the head.

File: ArduCopter/autoyaw.cpp (hunk 1 above)
Comment (tajisoft) [reply ArduCopter/autoyaw.cpp:126]: @muramura Thank you so much for your advise. And I'm sorry about the delay on getting back to this. I try to understand your point.

**Comment by tatsuy** [comment]
@tajisoft 
the angular speed of MAV_CMD_CONDITION_YAW is supported from Copter 4.1.
You can close this if it meets your requirements.

**Comment by tajisoft** [comment]
@tatsuy Thanks!!

**Pull request closed without merging**